libvast/src/system/meta_index.cpp

```diff
@@ -198,16 +199,17 @@ std::vector<uuid> meta_index_state::lookup_impl(const expression& expr) const {
     [&](const predicate& x) -> result_type {
       // Performs a lookup on all *matching* synopses with operator and
       // data from the predicate of the expression. The match function
-      // uses a qualified_record_field to determine whether the synopsis should
-      // be queried.
+      // uses a qualified_record_field to determine whether the synopsis
+      // should be queried.
       auto search = [&](auto match) {
         VAST_ASSERT(caf::holds_alternative<data>(x.rhs));
         const auto& rhs = caf::get<data>(x.rhs);
         result_type result;
         for (const auto& [part_id, part_syn] : synopses) {
           for (const auto& [field, syn] : part_syn.field_synopses_) {
             if (match(field)) {
-              auto cleaned_type = vast::legacy_type{field.type}.attributes({});
+              auto cleaned_type = field.type();
+              cleaned_type.prune_metadata();
```

Comment:
This is because type synopses only work for basic types. We want to add a function like this for that purpose:

```cpp
friend auto basic_type(const type& in) -> std::optional<type> {
  auto f = detail::overload{
    []<basic_type T>(const T& out) -> std::optional<type> {
      return out;
    },
    []<complex_type T>(const T&) -> std::optional<type> {
      return std::nullopt;
    },
  };
  return caf::visit(f, in);
};
```